### ISSUE TITLE
E208: Replace module should not trigger rule violation

### ISSUE THREAD

**Issue by aston-r** [ansible/ansible-lint#1023]
```
ansible-lint --version
ansible-lint 4.3.3
```

I have a question regarding the '[208] File permissions not mentioned' check for replace module.  Ansible-lint v4.3.0+ checks whether the mode is specified for replace module, but I have two concerns:

- replace module **_does not_** support 'mode: preserve' (I checked this with the newest version of ansible)
- replace module uses atomic_move function and permissions are preserved by default:
```
    def atomic_move(self, src, dest, unsafe_writes=False):
....
        if os.path.exists(b_dest):
            try:
                dest_stat = os.stat(b_dest)
                # copy mode and ownership
                os.chmod(b_src, dest_stat.st_mode & PERM_BITS)
                os.chown(b_src, dest_stat.st_uid, dest_stat.st_gid)
```

I actively use the replace the module and specifying the numeric mode(aka 0640,0755,0600) in the playbooks looks as overkill considering that the module preserves the permission by default. 

Fixes: #1024

**Comment by ssbarnea** [comment]
https://docs.ansible.com/ansible/latest/modules/replace_module.html#parameter-mode

**Comment by aston-r** [comment]
@ssbarnea 

I mentioned in the tt description that replace module does not support 'mode: preserve'

**Comment by ssbarnea** [comment]
Documentation states that it does support preserve. File a bug with ansible first. From the linter point of view, documentation is the main source of truth, not the current implementation inside (which we know very well that it can change over night, like it did with the permissions change that caused us to create this rule).

**Comment by aston-r** [comment]
@ssbarnea 

Thank you for your reply. Other people already mentioned this issue in ansible repo: https://github.com/ansible/ansible/issues/56839#issuecomment-677773269

and here is a reply from Matt Martz(Ansible Core Engineering Technical Team Lead): https://github.com/ansible/ansible/issues/56839#issuecomment-680082091

The documentation is wrong and will be fixed.

**Comment by ssbarnea** [comment]
Thanks! If you can also propose a PR to address it I will be happy to help you ship the fix.
